### PR TITLE
Feature explainer: Error causes

### PR DESCRIPTION
**Pull request by victorgomes**
Adds a feature explainer post about error causes.

### CODE REVIEW

File: src/features/error-cause.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      handleSomeWorkFailure(err.cause);
+      break;
+    case 'More work failed':
+      handleMoreWorkFailure(err.cause);
+      break;
+  }
+}
+```
+
+This feature is available in V8 v9.3.
```
Comment (mathiasbynens) [inline src/features/error-cause.md]: WDYT about adding a feature support section? See the other feature explainers for examples.

**Review by mathiasbynens — APPROVED** [review]

File: src/features/error-cause.md (hunk 1 above)
Comment (victorgomes) [reply src/features/error-cause.md]: Done. Thanks.

**Comment by mathiasbynens** [comment]
Awesome write-up. Thanks, Victor

 @rreverser, could you PTAL and help get this published?

**Comment by RReverser** [comment]
Spacing this one out a bit away from Object.hasOwn.

**Pull request merged**